### ISSUE TITLE
PHP Notice:  unserialize(): Error at offset (jobs, job queue)

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticMediaWiki#1033]
Setup: MW 1.23.9, SMW 2.2.0, PHP 5.4.39, MySQL 5.5.43

Cron spit out the following notice and warnings today:
```text
PHP Notice:  unserialize(): Error at offset 65535 of 65535 bytes in /.../includes/jobqueue/JobQueueDB.php on line 817
PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /.../extensions/SemanticMediaWiki/src/MediaWiki/Jobs/JobBase.php on line 107
PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /.../extensions/SemanticMediaWiki/src/MediaWiki/Jobs/JobBase.php on line 107
```
The problem here is, that I do not have access to the wiki in question. Perhaps this is enough to make use of it. Otherwise we can just keep it as a reminder (closed state) in case a similar or identical case pops up.

@kghbln internal reference 39992/2015-05-21T09:35

**Comment by mwjames** [comment]
The line in question is [0] because we always assumed that MW would send use an array, so instead of just trusting MW with:

```
return isset( $this->params[$key] ) || array_key_exists( $key, $this->params );
```
do an extra check similar to
```
if ( !is_array( $this->params ) ) {
 return false;
}

return isset( $this->params[$key] ) || array_key_exists( $key, $this->params );
```

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/src/MediaWiki/Jobs/JobBase.php#L107

**Comment by kghbln** [comment]
Oh, that's fast. :) Should I do a PR adding the extra check mentioned. Not sure about the "similar" though.

**Comment by mwjames** [comment]
> Should I do a PR adding the extra check mentioned.

:+1: 

> Not sure about the "similar" though.

I was just prose.

**Comment by kghbln** [comment]
@mwjames Thanks a ton!

**Comment by kghbln** [comment]
@mwjames This pull request indeed fixed the warnings, the notice however is still appearing. 

```text
PHP Notice:  unserialize(): Error at offset 65494 of 65535 bytes in /.../includes/jobqueue/JobQueueDB.php on line 817
```
This is probably a thing for core MediaWiki?

**Comment by mwjames** [comment]
>  This pull request indeed fixed the warnings (SMW 2.2.1), the notice however is still appearing.

`JobQueueDB.php` is managed by he master-disaster-recovery-mw-development-crew. So, yes this should be reported via Pabricator.

**Comment by kghbln** [comment]
> So, yes this should be reported via Pabricator.

Thanks for the reassurance! Done with [T101257](https://phabricator.wikimedia.org/T101257).
